docs/design/multi-node-maintenance.md

```
@@ -0,0 +1,229 @@
+
+This document introduces a new component within NICo-core which allows execution of multi-node maintenance operations in a conflict-free and efficient manner.
+
+Multi-Node maintenance executor (MNME) is a replacement for the current maintenance execution via the rack-state machine. It improves over it in the following areas:
+- MNME allows users to apply multiple maintenance operations to different trays within rack concurrently as long as it determines that there are no side-effects between the operations.
+- MNME allows to express safe maintenance operations for systems where maintenance side-effects do not exactly match rack boundaries. It supports architectures where the rack is no longer equivalent to the scaleup domain. That includes systems where
+  - the maintenance domain (e.g. NVLink failure domain) is smaller than a rack - e.g. if a GB200 rack is subdivided into multiple "mini racks"
+  - the maintenance domain spans multiple racks due to a any kind of shared resources
+
+Implementing MNME moves most logic out of the Rack State Machine into a different component, which reduces the need for having a type visible in NICo (`message Rack`) which does not actually physically exist.
+
+### What are multi-node maintenance operations?
```

markdownlint-cli2 (0.23.2)

[warning] Heading levels should only increment by one level at a time
Expected: h2; Actual: h3

(MD001, heading-increment)